### ISSUE TITLE
The future of knife-solo

### ISSUE THREAD

**Issue by tmatilai** [matschaffer/knife-solo#297]
One big step towards deprecating chef-solo was [taken](https://twitter.com/jkeiser2/status/388460927026085888) [yesterday](https://twitter.com/jkeiser2/status/388462998085316608) when `chef-client --local-mode` made it into Chef's master branch. The option starts a local [chef-zero](https://github.com/jkeiser/chef-zero) server which reads cookbooks, roles, nodes, data bags, etc. from the file system hierarchy ("kitchen"). It also supports search, and saving the node back to disk.

Sounds cool, doesn't it? =)

But this opens the question what will happen to knife-solo. (Another opening about the future was in #246, but in my opinion there is still need for provisioning servers without Vagrant :))

My earlier idea was to run chef-zero on the workstation, and then configure chef-client on the node to use it through an SSH tunnel. But synching the kitchen and running `chef-client --local-mode` sounds much simpler.

There are still some tasks that are needed:
1. Installing Chef to the node
2. Running the cookbook manager locally (Berkshelf, Librarian-Chef)
3. Uploading the kitchen with rsync (or pulling it from e.g. git)
4. Running chef-client on the node
5. Updating changes to the local kitchen (or to git)

I don't even see much need for Knife any more. A better fit could be integration with the next gen [bootstrapper](https://github.com/opscode/bootstrapper).

I'm interested to hear what @matschaffer and everyone else are thinking. Pinging also some involved Opscode staff if you already have some plans: @jkeiser, @danielsdeleo, @lamont-granquist

**Comment by allaire** [comment]
@matschaffer @tmatilai Is there something similar to knife solo for chef zero? Or plan to support it? Love the fact that I can do everything via SSH (à la Ansible) for smaller projects instead of spinning a Chef server.

**Comment by matschaffer** [comment]
@allaire I think switching knife-solo to chef-zero would be great and not too difficult. But I'm mainly using ansible and puppet at the moment so haven't had much need to work on this. Could probably pick it up a bit later in the spring but I'm happy to accept contributors in the mean time. I'm available in #knife-solo on freenode for questions any time if you're interested.

**Comment by kyle-johnson** [comment]
Relevant? https://github.com/higanworks/knife-zero

**Comment by matschaffer** [comment]
Definitely. I haven't really used chef in a few years beyond maintaining this project so hard to say when I'll pick it up.

Though I have some friends using knife-solo here so I might do it for their sake when time allows.
